File: leetcode/array/medium/java_solutions/Search2DMatrix_74.java
package leetcode.array.medium.java_solutions;

public class Search2DMatrix_74 {

    private static class Solution {

        public boolean searchMatrix(int[][] matrix, int target) {
            int N = matrix.length, M = matrix[0].length;

            int row = 0, col = M - 1;

            while (row < N && col >= 0) {
                int current = matrix[row][col];

                if (current < target) {
                    row++;
                } else if (current > target) {
                    col--;
                } else {
                    return true;
                }
            }
            return false;
        }
    }
}

File: leetcode/array/medium/golang_solutions/search_2d_matrix_74.go
package golang_solutions

func searchMatrix(matrix [][]int, target int) bool {
	row, col := 0, len(matrix[0])-1
	for row < len(matrix) && col >= 0 {
		if matrix[row][col] > target {
			col--
		} else if matrix[row][col] < target {
			row++
		} else {
			return true
		}
	}
	return false
}
